src/libraries/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs

```diff
@@ -416,11 +416,15 @@ public float GetSingle()
         internal float GetSingleWithQuotes()
         {
             ReadOnlySpan<byte> span = GetUnescapedSpan();
-            if (!TryGetSingleCore(out float value, span))
+
+            char numberFormat = JsonReaderHelper.GetFloatingPointStandardParseFormat(span);
```

Comment:
So we can't use (or extend) the raw reader here -- instead we parse format first.

Would it be better (performance or error-checking) to modify the reader code to support this directly?

Reply:
> Would it be better (performance or error-checking) to modify the reader code to support this directly?

Yes it would probably would be better for both performance & error checking. I did consider extending the raw reader to have numeric support in the `JsonTokenType.String` mode, but that is turning out to be a bit time consuming, and will increase the test matrix and validation for this already large PR. I plan to log this as an issue and work on it independently of this feature.

wrt. error checking, I feel pretty good about the test coverage/validation, and with performance, this is a new feature which doesn't hurt existing perf scenarios, so I'm okay with getting to this later.

> instead we parse format first

@Jozkee pointed out earlier that we technically don't have to pass the standard format to support the scientific notation, but I have it this way to be consistent with the handling in the `JsonTokenType.Number` [code paths](https://github.com/dotnet/runtime/blob/bfb01c942038e85e9853764d8f8940f9acb025e3/src/libraries/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs#L1496).

Reply:
Yes we could look at changing the reader later. Basically to avoid the double-parse for internal code where we have a string token but know it should be a number within. We may not be able to make this same optimization (defer the reader validation) for public code since it will change semantics.